storage/client/blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -121,29 +129,77 @@ private AzureBlobStorageBuilder buildImpl() {
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
      */
-    public BlobAsyncClient buildAsyncClient() {
+    public BlobAsyncClient buildBlobAsyncClient() {
         return new BlobAsyncClient(buildImpl(), snapshot);
     }
 
+    /**
+     * @return a {@link AppendBlobClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public AppendBlobClient buildAppendBlobClient() {
+        return new AppendBlobClient(buildAppendBlobAsyncClient());
+    }
+
+    /**
+     * @return a {@link AppendBlobAsyncClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public AppendBlobAsyncClient buildAppendBlobAsyncClient() {
+        return new AppendBlobAsyncClient(buildImpl(), snapshot);
+    }
+
+    /**
+     * @return a {@link BlockBlobClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public BlockBlobClient buildBlockBlobClient() {
+        return new BlockBlobClient(buildBlockBlobAsyncClient());
+    }
+
+    /**
+     * @return a {@link BlockBlobAsyncClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public BlockBlobAsyncClient buildBlockBlobAsyncClient() {
+        return new BlockBlobAsyncClient(buildImpl(), snapshot);
+    }
+
+    /**
+     * @return a {@link PageBlobClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public PageBlobClient buildPageBlobClient() {
+        return new PageBlobClient(buildPageBlobAsyncClient());
+    }
+
+    /**
+     * @return a {@link PageBlobAsyncClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
+     */
+    public PageBlobAsyncClient buildPageBlobAsyncClient() {
+        return new PageBlobAsyncClient(buildImpl(), snapshot);
+    }
+
     /**
      * Sets the service endpoint, additionally parses it for information (SAS token, container name, blob name)
      * @param endpoint URL of the service
      * @return the updated BlobClientBuilder object
-     * @throws IllegalArgumentException If {@code endpoint} is a malformed URL.
+     * @throws IllegalArgumentException If {@code endpoint} is {@code null} or is a malformed URL.
      */
     public BlobClientBuilder endpoint(String endpoint) {
```

Comment:
This method needs to be able to accept a fully formed URL to a blob resource. This only takes in the scheme, host, snapshot, and SAS; the path is never looked at.

Reply:
Good catch, I'll clean that up.

Reply:
Turns out that `URLParser` which this call into handles splitting the path into containerName and blobName.

Reply:
Turns out the path is handled by the URLParser, it is turned into containerName and blobName pieces.